Client/ZeebeClient.cs

```diff
@@ -122,8 +122,20 @@ public IPublishMessageCommandStep1 NewPublishMessageCommand()
 
         public void Dispose()
         {
-            gatewayClient = new ClosedGatewayClient();
-            channelToGateway.ShutdownAsync().Wait();
+            try
+            {
+                if (gatewayClient is ClosedGatewayClient)
+                {
+                    return;
+                }
+
+                gatewayClient = new ClosedGatewayClient();
+                channelToGateway.ShutdownAsync().Wait();
+            }
+            catch (System.Exception)
+            {
+                // ignore
```

Comment:
Please lets not ignore exceptions. Is there a specific reason for that?

Reply:
Because `Dispose()` should never throw exceptions, and there's no guarantee that `ShutdownAsync` won't throw here (in fact there's a fair chance it actually will throw...)

Reply:
That said; If you'd rather have the exception be thrown here, I can remove the try-catch. The real fix is the guard above that will make sure we're not calling `ShotdownAsync()` on a `ClosedGatewayClient` (which is guaranteed to throw an exception)

Reply:
yeah lets remove it